pandas/io/pytables.py

```diff
@@ -331,11 +331,17 @@ def read_hdf(path_or_buf, key=None, **kwargs):
 
     try:
         if key is None:
-            keys = store.keys()
-            if len(keys) != 1:
-                raise ValueError('key must be provided when HDF file contains '
-                                 'multiple datasets.')
-            key = keys[0]
+            groups = store.groups()
+            candidate_only_group = groups[0]
```

Comment:
I think this will fail on an empty store (and the error it raises will be odd, e.g. ``IndexError``).